airflow/serialization/serialized_objects.py

```diff
@@ -195,6 +196,24 @@ def serialize_to_json(
                 serialized_object[key] = value
         return serialized_object
 
+    @classmethod
+    def serialize_timezone(cls, var: datetime.tzinfo) -> Any:
+        """Serializes a timezone to json"""
+        tz_name = None
+        try:
+            tz_name = var.name
+        except AttributeError:
+            pass
+        if not tz_name:
+            try:
+                tz_name = var.tzname(datetime.datetime.utcnow())
+            except InvalidTimezone:
+                pass
+        if not tz_name:
+            raise TypeError(f"the timezone {var} can't be serialized")
```

Comment:
```suggestion
        tz_name = getattr(var, 'name', None)
        if not tz_name:
            try:
                tz_name = var.tzname(datetime.datetime.utcnow())
            except InvalidTimezone:
                raise TypeError(f"the timezone {var} can't be serialized")

```

This should have the same behaviour with less conditionals, right?